q/q.go

```diff
@@ -348,36 +348,39 @@ Query:
 		}
 		r, rtt, e := c.Exchange(m, nameserver)
 	Redo:
-		if e != nil {
+		switch e {
+		case nil:
+			//do nothing
+		case dns.ErrTruncated:
+			if *fallback {
+				if !*dnssec {
+					fmt.Printf(";; Truncated, trying %d bytes bufsize\n", dns.DefaultMsgSize)
+					o := new(dns.OPT)
+					o.Hdr.Name = "."
+					o.Hdr.Rrtype = dns.TypeOPT
+					o.SetUDPSize(dns.DefaultMsgSize)
+					m.Extra = append(m.Extra, o)
+					r, rtt, e = c.Exchange(m, nameserver)
+					*dnssec = true
+					goto Redo
+				} else {
+					// First EDNS, then TCP
+					fmt.Printf(";; Truncated, trying TCP\n")
+					c.Net = "tcp"
+					r, rtt, e = c.Exchange(m, nameserver)
```

Comment:
hmm, by tcp, the msg will cut into slices, so do we should consider about this case? Maybe I am wrong, if so, I am really sorry~